pkg/storage/cache.go

```diff
@@ -200,23 +229,63 @@ func (c *objectCache) storeAll(objs []meta.Object) {
 	}
 }
 
+func (c *objectCache) storeMeta(obj meta.Object) {
+	log.Printf("cache: storing %s meta object\n", obj.GetKind())
+	c.storeCO(cacheObject{object: obj, apiType: true})
+}
+
+func (c *objectCache) storeAllMeta(objs []meta.Object) {
+	for _, obj := range objs {
+		if uids, ok := c.objects[obj.GetKind()]; ok {
+			if _, ok := uids[obj.GetUID()]; ok {
+				continue
+			}
+		}
+
+		c.storeMeta(obj)
+	}
+}
+
 func (c *objectCache) delete(kind meta.Kind, uid meta.UID) {
 	if uids, ok := c.objects[kind]; ok {
 		delete(uids, uid)
 	}
 }
 
 func (c *objectCache) count(kind meta.Kind) uint64 {
-	return uint64(len(c.objects[kind]))
+	count := uint64(len(c.objects[kind]))
+	log.Printf("cache: counted %d %s objects\n", count, kind)
```

Comment:
in a follow-up, might want to remove this

Reply:
Yes, that's removed along with all other debug prints when I get to verifying all the functionality